### ISSUE TITLE
Slow Device To Host Transfers

### ISSUE THREAD

**Issue by MikeynJerry** [pytorch/xla#5188]
## ❓ Questions and Help

Recently I tried ResNet-50 on TPUs using this repo and TensorFlow / Keras. The performance difference between the two was about 15% (2844.4 img/s per TPU vs 3283.52 img/s) in favor of TensorFlow / Keras. These results were with logging every _300_ iterations. When I removed the logging, the TensorFlow / Keras performance stayed the same while this repo caught up within a few percent (3193.6 img/s). I think this is somewhat expected, as in a previous issue and in the troubleshooting guide, these transfers are generally seen as bad for performance. However, TensorFlow / Keras didn't have a change in their performance, so I did some digging, and it seems they use a separate thread and [device-specific outfeed queue](https://github.com/tensorflow/estimator/blob/7d846da87ed70f9a6c21a33a1c7178697844d9c0/tensorflow_estimator/python/estimator/tpu/tpu_estimator.py#LL450C19-L450C19) that lets them asynchronously transfer data (like the loss) to the host and display a progress bar and other metrics without any hit to TPU performance. Is there a reason they're able to do that and PyTorch XLA cannot?

More details:
- [PyTorch XLA script](https://github.com/pytorch/xla/blob/master/test/test_train_mp_imagenet.py) (taken from this repo's tests)
- [TensorFlow / Keras script](https://github.com/tensorflow/tpu/blob/master/models/experimental/resnet50_keras/resnet50.py) (taken from tensorflow/tpu)
- Performance statistics were taken in the exact same way across both scripts by measuring time right after each step (includes data loading time)
- For torch_xla, `add_step_closure` was tried with `run_async=False` and `run_async=True`.
- Both were run on the exact same v4-8 TPU VM with the latest version of torch_xla and tensorflow.

**Comment by JackCaoG** [comment]
@chandrasekhard2 since you has been looking into the Resnet benchmark, can you take a look?
